lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -102,17 +167,23 @@ static LogicalResult checkBody(linalg::LinalgOp linalgOp) {
 static FailureOr<SmallVector<Value>>
 getSlicedOperands(OpBuilder &builder, Location loc, ValueRange localIvs,
                   linalg::LinalgOp linalgOp, ValueRange valuesToUse) {
-  assert(linalgOp->getNumOperands() == 3 &&
-         "expect 3 input/output operands");
+  assert(linalgOp->getNumOperands() == 3 && "expect 3 input/output operands");
   assert(linalgOp.getDpsInputOperands().size() == 2 &&
          "expect 2 input operands");
 
   SmallVector<Value> slicedOperands;
   for (OpOperand *operand : linalgOp.getDpsInputOperands()) {
-    FailureOr<Value> slicedOperand = utils::getSliceOperand(
-        builder, operand, linalgOp, localIvs, valuesToUse, 3);
-    if (failed(slicedOperand))
-      return failure();
+    FailureOr<Value> slicedOperand;
+    AffineMap map = linalgOp.getMatchingIndexingMap(operand);
+    if (map.getNumResults() == 5) {
```

Comment:
This if switches on VNNI brgemm and traditional BRGEMM? If this is the case I don't think is enough.

Reply:
This if checks if the op's map has result == 5 implying that the operand is packed for VNNI. I have added a flag to cover the linalg op check for VNNI. I can't think of anything more to check, can you point out what other check might have to be covered?